core/iwasm/aot/debug/jit_debug.c

```diff
@@ -0,0 +1,224 @@
+ * limitations under the License.
+ * 
+ * Copyright (C) 2021 Ant Group.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "bh_log.h"
+#include "bh_platform.h"
+#include "wasm_runtime.h"
+
+#include <stdio.h>
+#include <assert.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <stdbool.h>
+
+// This must be kept in sync with gdb/gdb/jit.h .
+#ifdef __cplusplus
+extern "C" {
+#endif
+typedef enum {
+    JIT_NOACTION = 0,
+    JIT_REGISTER_FN,
+    JIT_UNREGISTER_FN
+} JITAction;
+typedef struct JITCodeEntry {
+    struct JITCodeEntry *next_;
+    struct JITCodeEntry *prev_;
+    const uint8_t *symfile_addr_;
+    uint64_t symfile_size_;
+} JITCodeEntry;
+
+typedef struct JITDescriptor {
+    uint32_t version_;
+    uint32_t action_flag_;
+    JITCodeEntry *relevant_entry_;
+    JITCodeEntry *first_entry_;
+} JITDescriptor;
+
+//LLVM has already define this.
+#if (WASM_ENABLE_WAMR_COMPILER == 0) && (WASM_ENABLE_JIT == 0)
+// GDB will place breakpoint into this function.
+// To prevent GCC from inlining or removing it we place noinline attribute
+// and inline assembler statement inside.
+void __attribute__((noinline)) __jit_debug_register_code();
+
+void __attribute__((noinline)) __jit_debug_register_code()
+{
+    int x;
+    *(char *)&x = '\0';
+}
+
+// GDB will inspect contents of this descriptor.
+// Static initialization is necessary to prevent GDB from seeing
+// uninitialized descriptor.
+
+JITDescriptor __jit_debug_descriptor = { 1, JIT_NOACTION, NULL, NULL };
+#else
+extern void __jit_debug_register_code();
+extern JITDescriptor __jit_debug_descriptor;
+#endif
+
+// Call __jit_debug_register_code indirectly via global variable.
+// This gives the debugger an easy way to inject custom code to handle the events.
+void (*__jit_debug_register_code_ptr)() = __jit_debug_register_code;
+
+#ifdef __cplusplus
+}
+#endif
+
+typedef struct WASMJITDebugEngine {
+    korp_mutex jit_entry_mutex;
+    bh_list jit_entry_list;
+} WASMJITDebugEngine;
+
+typedef struct WASMJITEntryNode {
+    struct WASMJITEntryNode *next;
+    ;
+    JITCodeEntry *entry;
+} WASMJITEntryNode;
+
+static WASMJITDebugEngine *jit_debug_engine;
+
+static WASMJITDebugEngine *
+wasm_jit_debug_engin_create()
+{
+    WASMJITDebugEngine *engine;
+    if (!(engine = wasm_runtime_malloc(sizeof(WASMJITDebugEngine)))) {
+        LOG_ERROR("WASM Debug Engine error: failed to allocate memory");
+        return NULL;
+    }
+    memset(engine, 0, sizeof(WASMJITDebugEngine));
+    return engine;
+}
+
+void
+initial_jit_debug_engine()
+{
+    if (!jit_debug_engine)
+        jit_debug_engine = wasm_jit_debug_engin_create();
+    os_mutex_init(&jit_debug_engine->jit_entry_mutex);
+    bh_list_init(&jit_debug_engine->jit_entry_list);
+}
+
+static JITCodeEntry *
+CreateJITCodeEntryInternal(const uint8_t *symfile_addr, uint64_t symfile_size)
```

Comment:
Had better not use uint8_t, uint32_t, uint64_t inside vm core, use uint8/uint32/uint64 instead.